File: StudyEnglishJava/src/main/java/org/example/studyenglishjava/dto/IStudyType.java
package org.example.studyenglishjava.dto;

public interface IStudyType {

    int value();

    public enum Learn implements IStudyType {
        INSTANCE;

        @Override
        public int value() {
            return 1;
        }
    }

    public enum Review implements IStudyType {
        INSTANCE;

        @Override
        public int value() {
            return 2;
        }
    }
}

File: LearningEnglish/app/src/main/java/com/example/learningenglish/word/model/IStudyType.kt
package com.example.learningenglish.word.model

sealed interface IStudyType {

    fun value(): Int

    data object Learn : IStudyType {
        override fun value(): Int {
            return 1
        }
    }

    data object Review : IStudyType {
        override fun value(): Int {
            return 2
        }
    }

}